### PR TITLE
Escalation policy card

### PR DESCRIPTION
**Pull request by mjholder**
this adds escalation policy info for on-call purposes. It provides the description for the policy and links for each form of contact for the teams responsible.

### CODE REVIEW

File: plugins/visual-qontract/src/components/EscalationPolicyComponent/NextEscalationPolicyRow.tsx [hunk 1]
```diff
@@ -0,0 +1,82 @@
...
+      <TableCell width="25%">
+        <Grid item>
+          <Grid container direction="column">
+            <Grid item>
+              <Typography >
+                <Box sx={{ alignItems: "baseline" }}>
+                  email:
+                  {" "}
+                  <Link target="_blank" href={"mailto:"+ep.channels.email}>
+                    {ep.channels.email}
```
Comment (adamrdrew) [inline plugins/visual-qontract/src/components/EscalationPolicyComponent/NextEscalationPolicyRow.tsx:48]: This code makes me a bit nervous. You are referring in JSX to nested properties of an object that comes from a data source you don't control. What if one of these properties is missing? Or, what if the `ep.channels.slackUserGroup[0]` array is empty? You are trying to call `ep.channels.slackUserGroup[0].handle`. There's a lot of cases here where this JSX could fail to render and the component could bomb out. I think you need to refactor this to safely pull the properties you want out, and handle cases where stuff isn't there.

File: plugins/visual-qontract/src/components/EscalationPolicyComponent/NextEscalationPolicyRow.tsx [hunk 2]
```diff
@@ -0,0 +1,82 @@
...
+              </Typography>
+            </Grid>
+            { ep.channels.slackUserGroup[0].handle && 
+            <Grid item>
+              <Typography>
+                <Box sx={{ alignItems: "baseline" }}>
+                  Slack:
+                  {" "}
+                  <Link target="_blank" href={getAppInterfaceLink(ep.channels.slackUserGroup[0].path)}>
+                    {ep.channels.slackUserGroup[0].handle}
```
Comment (adamrdrew) [inline plugins/visual-qontract/src/components/EscalationPolicyComponent/NextEscalationPolicyRow.tsx:60]: See above comment - this is unsafe.

**Review by adamrdrew — CHANGES_REQUESTED** [review]
Great contribution! I love this component. You've got a bit more work to do here to refine things, but this is darn close!

File: plugins/visual-qontract/src/components/EscalationPolicyComponent/EscalationPolicyComponent.tsx [hunk 3]
```diff
@@ -0,0 +1,134 @@
...
+            setNextPath(data.escalation_policies_1[0].channels.nextEscalationPolicy.path);
+          }
+          if (escalationPolicies.some(e => e.path === data.escalation_policies_1[0].path)) {
+            return;
+          }
+          setEscalationPolicies([...escalationPolicies, data.escalation_policies_1[0]]);
+        }
+      })
+      .catch((_error) => {
+        console.log("no work")
```
[the remark this answers is not in the document]
Comment (mjholder) [reply plugins/visual-qontract/src/components/EscalationPolicyComponent/EscalationPolicyComponent.tsx]: Oof, yeah I have a problem with leaving bad log messages around.

File: plugins/visual-qontract/src/components/EscalationPolicyComponent/NextEscalationPolicyRow.tsx [hunk 4]
```diff
@@ -0,0 +1,82 @@
...
+import QueryQontract from '../../common/QueryAppInterface';
+
+
+const appInterfaceBaseURL = `https://gitlab.cee.redhat.com/service/app-interface/-/blob/master/data`;
+
+const getAppInterfaceLink = (path: string) => {
+  return `${appInterfaceBaseURL}${path}`;
+};
+
+export const EscalationPolicyRow = ({ep}: {ep: any}) => {
```
[the remark this answers is not in the document]
Comment (mjholder) [reply plugins/visual-qontract/src/components/EscalationPolicyComponent/NextEscalationPolicyRow.tsx]: oops! good catch

File: plugins/visual-qontract/src/components/EscalationPolicyComponent/NextEscalationPolicyRow.tsx (hunk 1 above)
Comment (mjholder) [reply plugins/visual-qontract/src/components/EscalationPolicyComponent/NextEscalationPolicyRow.tsx]: I refactored the email, slack and jira entries to be instances of a "ContactItem". It makes things easier to read and maintain. Along with that I added a few checks to make sure things exist before using them.

**Pull request merged**